amf-api-contract/shared/src/main/scala/amf/apicontract/internal/transformation/compatibility/oas/FixFilePayloads.scala

```diff
@@ -0,0 +1,86 @@
+package amf.apicontract.internal.transformation.compatibility.oas
+
+import amf.apicontract.client.scala.model.domain.Payload
+import amf.apicontract.client.scala.model.domain.api.WebApi
+import amf.apicontract.internal.annotations.FormBodyParameter
+import amf.core.client.scala.AMFGraphConfiguration
+import amf.core.client.scala.errorhandling.AMFErrorHandler
+import amf.core.client.scala.model.DataType
+import amf.core.client.scala.model.document.{BaseUnit, Document}
+import amf.core.client.scala.transform.TransformationStep
+import amf.shapes.client.scala.model.domain.{FileShape, NodeShape, ScalarShape}
+
+/** To represent a method with file upload:
+  *
+  * in RAML: \- You define a body with multipart/form-data and a property of type: file \- You can specify acceptable
+  * file types directly with fileTypes: ['application/xml']
+  *
+  * In OAS 2.0 you need a parameter of type: file and in: formData with a consumes: - multipart/form-data, to correctly
+  * emit this the FormBodyParameter annotation is needed
+  *
+  * In OAS 3.0: \- You define a requestBody with content of type multipart/form-data \- it's schema should be of type:
+  * object and have a property that uses type: string with format: binary
+  */
+trait FixFilePayloads extends TransformationStep {
+  override def transform(
+      model: BaseUnit,
+      errorHandler: AMFErrorHandler,
+      configuration: AMFGraphConfiguration
+  ): BaseUnit = {
+    model match {
+      case doc: Document if doc.encodes.isInstanceOf[WebApi] => fixFields(doc.encodes.asInstanceOf[WebApi])
+      case _                                                 => // ignore
+    }
+    model
+  }
+
+  private def fixFields(api: WebApi): Unit = {
+    api.endPoints.foreach { endPoint =>
+      endPoint.operations.foreach { operation =>
+        val requestPayloads  = operation.requests.flatMap(_.payloads)
+        val responsePayloads = operation.responses.flatMap(_.payloads)
+        val payloads         = requestPayloads ++ responsePayloads
+        payloads.foreach {
+          case p if hasFileUpload(p) => checkFormData(p)
+          case _                     => // ignore
+        }
+      }
+    }
+  }
+
+  private def hasFileUpload(payload: Payload): Boolean = {
+    payload.schema match {
+      case obj: NodeShape => obj.properties.exists(prop => prop.range.isInstanceOf[FileShape])
```

Comment:
Why check only for a NodeShape with a property file type? In the issue particular case the type was like this, but this is not always like that.
In RAML spec you have examples with file type at root level, and nothing limitates where it could be defined: https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#applying-libraries
The solution should be general for all possible cases